src/test/java/rx/SingleTest.java

```diff
@@ -1353,4 +1386,364 @@ public String call() throws Exception {
+        Subscription sOther = mock(Subscription.class);
+        TestSingle source = new TestSingle(sSource);
+        TestObservable other = new TestObservable(sOther);
+        Throwable error = new Throwable();
+
+        TestSubscriber<String> result = new TestSubscriber<String>();
+        Single<String> stringSingle = Single.create(source).takeUntil(Observable.create(other));
+        stringSingle.subscribe(result);
+        source.sendOnError(error);
+
+        result.assertError(error);
+        verify(sSource, times(1)).unsubscribe();
+        verify(sOther, times(1)).unsubscribe();
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testTakeUntilOtherError() {
+        Subscription sSource = mock(Subscription.class);
+        Subscription sOther = mock(Subscription.class);
+        TestSingle source = new TestSingle(sSource);
+        TestSingle other = new TestSingle(sOther);
+        Throwable error = new Throwable();
+
+        TestSubscriber<String> result = new TestSubscriber<String>();
+        Single<String> stringSingle = Single.create(source).takeUntil(Single.create(other));
+        stringSingle.subscribe(result);
+        other.sendOnError(error);
+
+        result.assertError(error);
+        verify(sSource, times(1)).unsubscribe();
+        verify(sOther, times(1)).unsubscribe();
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testTakeUntilOtherErrorObservable() {
+        Subscription sSource = mock(Subscription.class);
+        Subscription sOther = mock(Subscription.class);
+        TestSingle source = new TestSingle(sSource);
+        TestObservable other = new TestObservable(sOther);
+        Throwable error = new Throwable();
+
+        TestSubscriber<String> result = new TestSubscriber<String>();
+        Single<String> stringSingle = Single.create(source).takeUntil(Observable.create(other));
+        stringSingle.subscribe(result);
+        other.sendOnError(error);
+
+        result.assertError(error);
+        verify(sSource, times(1)).unsubscribe();
+        verify(sOther, times(1)).unsubscribe();
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testTakeUntilOtherCompleted() {
+        Subscription sSource = mock(Subscription.class);
+        Subscription sOther = mock(Subscription.class);
+        TestSingle source = new TestSingle(sSource);
+        TestObservable other = new TestObservable(sOther);
+
+        TestSubscriber<String> result = new TestSubscriber<String>();
+        Single<String> stringSingle = Single.create(source).takeUntil(Observable.create(other));
+        stringSingle.subscribe(result);
+        other.sendOnCompleted();
+
+        result.assertError(CancellationException.class);
+        verify(sSource, times(1)).unsubscribe();
+        verify(sOther, times(1)).unsubscribe();
+
+    }
+
+    private static class TestObservable implements Observable.OnSubscribe<String> {
```

Comment:
Most of these tests are adapted from the `OperatorTakeUntilTest`, which used these. I was just matching behavior. Could use subjects, but if we do then I think we should update both places.